gufe/custom_json.py

```diff
@@ -66,8 +66,8 @@ def default(self, obj: Any) -> Any:
             dct = {}
             if self.cls:
                 dct.update({
-                    '__class__': self.cls.__name__,
-                    '__module__': self.cls.__module__,
+                    '__class__': obj.__class__.__qualname__,
+                    '__module__': obj.__class__.__module__,
```

Comment:
What do these changes do? I think it broke the test:
```
FAILED gufe/tests/test_custom_json.py::TestPathCodec::test_default - AssertionError: assert {':is_custom:...h': 'foo/bar'} == {':is_custom:...h': 'foo/bar'}
  Omitting 3 identical items, use -vv to show
  Differing items:
  {'__class__': 'PosixPath'} != {'__class__': 'Path'}
  Full diff:
    {
     ':is_custom:': True,
  -  '__class__': 'Path',
  +  '__class__': 'PosixPath',
  ?                +++++
     '__module__': 'pathlib',
     'path': 'foo/bar',
    }
FAILED gufe/tests/test_custom_json.py::TestPathCodec::test_round_trip - AssertionError: assert {':is_custom:': True, '__class__': 'PosixPath', '__module__': 'pathlib', 'path': 'foo/bar'} == PosixPath('foo/bar')
FAILED gufe/tests/test_tokenization.py::test_path_to_json - AssertionError: assert PosixPath('foo/bar') == {':is_custom:': True, '__class__': 'PosixPath', '__module__': 'pathlib', 'path': 'foo/bar'}
FAILED gufe/tests/test_transformation.py::TestTransformation::test_dump_load_roundtrip - KeyError: '__class__'
```

Reply:
Commented [here](https://github.com/OpenFreeEnergy/gufe/pull/105#issuecomment-1344848538):

> Looks like, now that we're properly reporting the subclass, the tests for `pathlib.Path` should change a bit, too -- the new results are probably what we should have had from the beginning (specific subclass, not generic superclass)

Old version used the saved superclass, and would always recreate an instance of that, instead of the specific subclass. Obviously, we want to recreate a subclass, not the generic superclass.

Reply:
Got it! Now I understand what you mean by:

> the new results are probably what we should have had from the beginning